cli/slsa-verifier/verify/options.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verify
+
+import (
+	"fmt"
+	"strings"
+
+	serrors "github.com/slsa-framework/slsa-verifier/errors"
+	"github.com/spf13/cobra"
+)
+
+type Interface interface {
+	// AddFlags adds this options' flags to the cobra command.
+	AddFlags(cmd *cobra.Command)
+}
+
+// VerifyOptions is the top-level options for all `verify` commands.
+type VerifyOptions struct {
+	/* Source requirements */
+	SourceURI        string
+	SourceBranch     string
+	SourceTag        string
+	SourceVersionTag string
+	/* Builder Requirements */
+	BuildWorkflowInputs workflowInputs
+	BuilderID           string
+	/* Other */
+	ProvenancePath  string
+	PrintProvenance bool
+}
+
+var _ Interface = (*VerifyOptions)(nil)
+
+// AddFlags implements Interface
+func (o *VerifyOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&o.ProvenancePath, "provenance-path", "",
+		"path to a provenance file")
+
+	cmd.Flags().StringVar(&o.BuilderID, "builder-id", "", "EXPERIMENTAL: the unique builder ID who created the provenance")
+
+	cmd.Flags().StringVar(&o.SourceURI, "source-uri", "",
+		"expected source repository that should have produced the binary, e.g. github.com/some/repo")
+
+	cmd.Flags().StringVar(&o.SourceBranch, "source-branch", "", "[optional] expected branch the binary was compiled from")
+
+	cmd.Flags().StringVar(&o.SourceTag, "source-tag", "", "[optional] expected tag the binary was compiled from")
+
+	cmd.Flags().StringVar(&o.SourceVersionTag, "source-versioned-tag", "",
+		"[optional] expected version the binary was compiled from. Uses semantic version to match the tag")
+
+	cmd.Flags().BoolVar(&o.PrintProvenance, "print-provenance", false,
+		"print the verified provenance to stdout")
+
+	cmd.Flags().Var(&o.BuildWorkflowInputs, "build-workflow-input",
```

Comment:
btw, did you say you wanted to make this option `-build-input`?

Reply:
I thought simpler to include a migration from `workflow-input` to `build-workflow-input` like the `source` ones. i think `build-input` sounds better, but i'm hesitant it might mean builder inputs

Reply:
SGTM. Also, it makes it clearer it's for (GH) workflows.

Reply:
done